pkg/driver/controller.go

```diff
@@ -187,6 +194,26 @@ func (d *controllerService) CreateVolume(ctx context.Context, req *csi.CreateVol
 		}
 	}
 
+	if len(blockSize) > 0 {
+		for _, volCap := range req.GetVolumeCapabilities() {
+			switch volCap.GetAccessType().(type) {
+			case *csi.VolumeCapability_Block:
+				return nil, status.Error(codes.InvalidArgument, "Cannot use block size with block volume")
```

Comment:
I will test this and report back, but I believe it is just silently ignored. That feels really wrong here though, and I strongly believe we should raise an error - the user requested we set the block size in a situation where we definitely cannot.

Reply:
Update: it does silently fail, but this case is different, because (at least on k8s) the `external-attacher` _always_ injects the default fstype, even on block volumes, so we are incapable of detecting presence/absence.